keycloakClient/keycloakClient.go

```diff
@@ -117,6 +122,73 @@ func (c *Client) doTokenRequest(realm string, body string) (*TokenInfo, error) {
 	return tokenInfo, nil
 }
 
+func setAuthorizationAndHostHeaders(req *http.Request, accessToken string) (*http.Request, error) {
+	host, err := extractHostFromToken(accessToken)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", accessToken))
+	req.Header.Add("X-Forwarded-Proto", "https")
+	// is this always the content type? fails with 415 if content type isn't specified
+	req.Header.Add("Content-Type", "application/json")
```

Comment:
both of these could vary (e.g. for uploading a file it will be a different content type) so need to be parameters that get passed into this function